signal/signals.c

```diff
@@ -1,208 +1,5 @@
-#if defined(IOS_DEPLOYMENT)
-// ======================================================================================
-// iOS framework compilation using xgo
-// ======================================================================================
-
-#include <stddef.h>
-#include <stdbool.h>
-#include <stdlib.h>
-#include <objc/objc.h>
-#include <objc/runtime.h>
-#include <objc/message.h>
-
-static id statusServiceClassRef = nil;
-static SEL statusServiceSelector = nil;
-
-static bool initLibrary() {
-    if (statusServiceClassRef == nil) {
-        statusServiceClassRef = objc_getClass("Status");
-        if (statusServiceClassRef == nil) return false;
-    }
-
-    if (statusServiceSelector == nil) {
-        statusServiceSelector = sel_getUid("signalEvent:");
-        if (statusServiceSelector == nil) return false;
-    }
-
-    return true;
-}
-
-
-/*!
- * @brief Calls static method signalEvent of class GethService.
- *
- * @param jsonEvent - UTF8 string
- *
- * @note Definition of signalEvent method.
- *  + (void)signalEvent:(const char *)json
- */
-bool StatusServiceSignalEvent(const char *jsonEvent) {
-    if (!initLibrary()) return false;
-
-    void (*action)(id, SEL, const char *) = (void (*)(id, SEL, const char *)) objc_msgSend;
-    action(statusServiceClassRef, statusServiceSelector, jsonEvent);
-
-    return true;
-}
-
-void SetEventCallback(void *cb) {
-}
-
-#elif defined(ANDROID_DEPLOYMENT)
-// ======================================================================================
-// Android archive compilation using xgo
-// ======================================================================================
-
-#include <stddef.h>
-#include <stdbool.h>
-#include <jni.h>
-
-bool StatusServiceSignalEvent(const char *jsonEvent);
-
-static JavaVM *gJavaVM = NULL;
-static jclass JavaClassPtr_StatusService = NULL;
-static jmethodID JavaMethodPtr_signalEvent = NULL;
-
-static bool JniLibraryInit(JNIEnv *env);
-
-/*!
- * @brief Get interface to JNI.
- *
- * @return true if thread should be detached from JNI.
- */
-static bool JniAttach(JNIEnv **env) {
-	jint status;
-
-	if (gJavaVM == NULL) {
-		env = NULL;
-	}
-
-	status = (*gJavaVM)->GetEnv(gJavaVM, (void **)env, JNI_VERSION_1_6);
-	if (status == JNI_EDETACHED) {
-		// attach thread to JNI
-		//(*gJavaVM)->AttachCurrentThread( gJavaVM, (void **)env, NULL ); // Oracle JNI API
-		(*gJavaVM)->AttachCurrentThread(gJavaVM, env, NULL); // Android JNI API
-		return true;
-	} else if (status != JNI_OK) {
-		return false;
-	}
-
-	return false;
-}
-
-/*!
- * @brief The VM calls JNI_OnLoad when the native library is loaded.
- */
-JNIEXPORT jint JNI_OnLoad(JavaVM* vm, void* reserved) {
-	bool detach;
-	JNIEnv *env;
-	int result = JNI_VERSION_1_6;
-
-	gJavaVM = vm;
+// +build library,!android,!arm
```

Comment:
`!darwin` missing?

Reply:
No, `darwin` is a valid architecture name for MacOS X.